docs/guides/multi_engine.md

```diff
@@ -0,0 +1,112 @@
+# Multi-Engine guide
+
+Organizations typically connect to a data warehouse through a single engine to ensure data consistency. However, there are cases where the processing capabilities of one engine may be better suited to specific tasks than another.
+
+By decoupling storage from compute and with growing support for open table formats like Apache Iceberg and Hive, different engines can now interact with the same data.
+
+With SQLMesh's new multi-engine feature, users can leverage multiple engine adapters within a single project, offering the flexibility to choose the best engine for each task.
+
+This feature allows you to run each model on a specified engine, provided the data catalog is shared and the engines support read/write operations on it.
```

Comment:
sure so add maybe that it is possible to use a setup where the catalog is not shared, when the engine provides support or an integration for federated queries, but it would be advisable to proceed with caution in such cases?

Reply:
Perhaps, yeah. Let's see what others think about this use case.

Reply:
> What about use cases where the catalog's not shared but an engine provides native support / plugins for federated queries? 

Isnt this the existing single-gateway model, just against a federated query engine? Trino falls into this as well. You write your SQLMesh project against Trino and the fact that some catalogs are backed by another DB engine is an implementation detail. It's all the same SQL dialect, the query federation provides the compatibility

Reply:
I see, yeah I think your point is reasonable. Feel free to ignore this @themisvaltinos, perhaps we can even remove the final section in the markdown where we say "take caution [...]" and just leave the example there?

Reply:
ok removed this section